pyfr/plugins/fluidforce.py

```diff
@@ -117,10 +143,18 @@ def __call__(self, intg):
 
         # Solution matrices indexed by element type
         solns = dict(zip(intg.system.ele_types, intg.soln))
-        ndims, nvars = self.ndims, self.nvars
+        ndims, nvars, mcomp = self.ndims, self.nvars, self._mcomp
+
+        # Force and moment vectors
+        n = 2 if self._viscous else 1
+        fm = np.zeros((ndims + mcomp, n))
```

Comment:
Thinking out loud some more, would a 3d array work better (ndims, 1 + moments, 1 + visc) or some rearrangement thereof?

Reply:
This does not help much to clean the this code since it's difficult to unpack when writing the files. The main issue is that the moments may have different components than the force and therefore 2D/3D arrays are tricky. I think that the following may be cleaner...

```
fm = np.zeros((ndims + mcomp)*(1 + self._viscous))
f, m = np.split(fm, [ndims*(1 + self._viscous)])
fp, fv = np.split(f, 2)
mp, mv = np.split(m, 2)
...
print(intg.tcurr, *fm, sep=',', file=self.outf)
```